system/lib/compiler-rt/lib/asan/asan_emscripten.cc

```diff
@@ -0,0 +1,44 @@
+#include "asan_internal.h"
+#include "asan_mapping.h"
+#include "asan_poisoning.h"
+
+#if SANITIZER_EMSCRIPTEN
+#include <emscripten.h>
+#include <cstddef>
+
+namespace __asan {
+
+void InitializeShadowMemory() {
+  // Poison the shadow memory of the shadow area at the start of the address
+  // space. This helps catching null pointer dereference.
+  FastPoisonShadow(kLowShadowBeg, kLowShadowEnd - kLowShadowBeg, 0xff);
+}
+
+void AsanCheckDynamicRTPrereqs() {}
+void AsanCheckIncompatibleRT() {}
+void InitializePlatformInterceptors() {}
+void InitializePlatformExceptionHandlers() {}
+bool IsSystemHeapAddress (uptr addr) { return false; }
+
+void *AsanDoesNotSupportStaticLinkage() {
+  // Actually, we have to do static linkage on Emscripten.
```

Comment:
What is missing to enable dynamic linking, or have we just not explored that yet?

Reply:
Well, on Linux, this function is some sort of a bomb that is supposed to blow up on static linkage by doing some nasty things to break the static linker. Here, we make it do nothing.